### PR TITLE
[Java][okhttp-gson] validateJsonElement does not validate enum values

### LINKED ISSUE #16917: [BUG] [Java] [okhttp-gson] enum values are not validated in some cases
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

We are using the Java generator with okhttp-gson.  We have noticed that enum values are not validated consistently by the generated models.

With the [example schema below](#OpenAPIdeclarationfilecontentorurl), we have observed that the code generated by okhttp-gson generator works as follows when deserializing an object that matches the FooThing schema (i.e., it has a `kind` property with value `foo`):
- `FooThing` deserializes the object without an error (correct)
- `BarThing` throws an error that the object isn't valid according to the schema (correct)
- `Thing` model doesn't validate the enum values, throws an error that the object matches both FooThing and BarThing (incorrect)
- `Container` model, which only allows a `BarThing`, doesn't validate the enum value and does not throw a validation error (incorrect)

##### openapi-generator version

v7.0.0

##### OpenAPI declaration file content or url

```yaml
components:
  schemas:
    Thing:
      type: object
      oneOf:
        - $ref: '#/components/schemas/FooThing'
        - $ref: '#/components/schemas/BarThing'
    FooThing:
      type: object
      properties:
        kind:
          type: string
          enum:
          - foo
    BarThing:
      type: object
      properties:
        kind:
          type: string
          enum:
          - bar
    Container:
      type: object
      properties:
        thing:
          $ref: '#/components/schemas/BarThing'
```

##### Suggest a fix

Generate `validateJsonElement` functions when using the Java `okhttp-gson` generator

### PR DESCRIPTION
**Pull request by ctreatma**
This PR adds enum templates for Java with `okhttp-gson` to add `validateJsonElement` functions to both inline enums and external enums, and to call the `validateJsonElement` function for enum properties.

Fixes #16917

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc/ @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @martin-mfg (2023/08)

### CODE REVIEW

**Comment by wing328** [comment]
Please merge the latest master as I've just merged https://github.com/OpenAPITools/openapi-generator/pull/16985 to restore some tests.

Please add a test in JSONTest.java to ensure this issue is covered moving forward.

Let me know if you need any help.

**Comment by ctreatma** [comment]
@wing328 I added a test and it passed in CI but a different workflow failed because the digest of the `JSONTest.java` file changed: https://github.com/OpenAPITools/openapi-generator/actions/runs/6776363363/job/18417685804?pr=16865

Did I update the test file in the wrong way?  Or should I manually update the SHA in `bin/utils/test_file_list.yaml`?

**Comment by wing328** [comment]
> Or should I manually update the SHA in bin/utils/test_file_list.yaml?

yes please update that as the hash is no longer the same

File: modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/modelEnum.mustache [hunk 1]
```diff
@@ -0,0 +1,123 @@
+{{#jackson}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/modelEnum.mustache:1]: when you've time, can you please file another PR to clean up {{#jackson}} ... {{/jackson}} in this file as okhttp-gson library doesn't use the jackson mustache tag?

**Pull request merged**